client/src/components/TreeDropdownField/tests/TreeDropdownField-story.js

```diff
@@ -1,35 +1,50 @@
 import React from 'react';
-// eslint-disable-next-line import/no-extraneous-dependencies
+import { createStore, combineReducers } from 'redux';
+import { Provider } from 'react-redux';
 import { storiesOf } from '@storybook/react';
-// eslint-disable-next-line import/no-extraneous-dependencies
-// import { action } from '@storybook/addon-actions';
-// import { Component as TreeDropdownField } from 'components/TreeDropdownField/TreeDropdownField';
+import { withNotes } from '@storybook/addon-notes';
 import ValueTracker from 'stories/ValueTracker';
 
-// const triggerAction = action('action');
-/*
+import notes from '../README.md';
+import treeDropdownFieldReducer from 'state/treeDropdownField/TreeDropdownFieldReducer';
+import { ConnectedTreeDropdownField as TreeDropdownField } from 'components/TreeDropdownField/TreeDropdownField';
+import mockTree from './mockTree';
+
+// TreeDropdownField depends on having a basic Redux store present to operate
+const store = createStore(
+  combineReducers({ treeDropdownField: treeDropdownFieldReducer }),
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```

Comment:
is it reasonable to assume that everyone who would view this, including front-end designers, would have `__REDUX_DEVTOOLS_EXTENSION__` installed?

would it not just make more sense to create our own "basic Redux store"?

Reply:
The first part of this statement (`window.__REDUX_DEVTOOLS_EXTENSION__ &&`) ensures that it only runs if that property exists. This is fine. Also, it's a debug function that's provided by the redux devtools extension that should be run to hook up your redux store to the devtools extension, so this is still just a "basic Redux store".

This LGTM.

Reply:
yeah i mean it requires people viewing this to actually have that extension installed in their chrome/firefox.  I wouldn't expect a designer to have this installed

Reply:
Seeing the state changes for a component will almost certainly be irrelevant to designers,  and developers working with the state layer in Silverstripe CMS generally have the plugin installed for other diagnostic purposes, so I don't see any reason to try and bake it in at this stage. I'll add a note to the README stating that it's driven by Redux, though.

Reply:
Sorry, I was just saying I don't think "some people might not be using redux devtools" is a good reason to say "don't initialise redux devtools for anybody".